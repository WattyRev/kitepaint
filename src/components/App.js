import React from "react";
import { Route, Switch } from "react-router-dom";
import { isEmbedded } from "../utils/embed";
import Header from "./layout/Header";
import Footer from "./layout/Footer";
import RestrictedRoute from "./RestrictedRoute";
import About from "./about";
import Account from "./account";
import Activate from "./activate";
import Create from "./create";
import CreateNew from "./createNew";
import Designs from "./designs";
import Edit from "./edit";
import ErrorPage from "./ErrorPage";
import Home from "./home";
import MyDesigns from "./my-designs";
import View from "./view";

/**
 * The entry for the app. The router will go here.
 */
const App = () => (
  <React.Fragment>
    {!isEmbedded && <Header />}
    <Switch>
      <RestrictedRoute exact path="/" component={Home} />
      <RestrictedRoute exact path="/about" component={About} />
      <RestrictedRoute exact path="/account" component={Account} />
      <RestrictedRoute
        exact
        path="/activate/:userId/:activationCode"
        component={Activate}
      />
      <RestrictedRoute exact path="/create" component={Create} />
      <Route exact path="/create/:productId" component={CreateNew} />
      <RestrictedRoute exact path="/designs" component={Designs} />
      <RestrictedRoute exact path="/edit/:designId" component={Edit} />
      <RestrictedRoute exact path="/my-designs" component={MyDesigns} />
      <Route exact path="/view/:designId" component={View} />
      <Route component={ErrorPage} />
    </Switch>
    <Footer />
  </React.Fragment>
);

export default App;
