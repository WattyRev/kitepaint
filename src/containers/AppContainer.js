import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { CHECK_LOGIN } from "../redux/actions";
import { getCheckingLogin } from "../redux/modules/user";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faSignOutAlt } from "@fortawesome/free-solid-svg-icons";

library.add(faSignOutAlt);

class AppContainer extends React.Component {
  static propTypes = {
    children: PropTypes.node,
    isCheckingLogin: PropTypes.bool.isRequired,
    onCheckLogin: PropTypes.func.isRequired
  };

  constructor(props, ...rest) {
    super(props, ...rest);
    props.onCheckLogin();
  }

  render() {
    if (this.props.isCheckingLogin) {
      return "";
    }
    return this.props.children;
  }
}

const mapStateToProps = state => ({
  isCheckingLogin: getCheckingLogin(state)
});

const mapDispatchToProps = {
  onCheckLogin: CHECK_LOGIN
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AppContainer);
