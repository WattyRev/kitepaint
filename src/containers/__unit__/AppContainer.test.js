import React from "react";
import { shallow } from "enzyme";
import App from "../../components/App";
import { AppContainer } from "../AppContainer";

describe("AppContainer", () => {
  let props;
  beforeEach(() => {
    props = {
      isCheckingLogin: false,
      onCheckLogin: jest.fn(() => new Promise(resolve => resolve()))
    };
  });
  it("renders", () => {
    shallow(<AppContainer {...props} />);
  });
  describe(".props", () => {
    describe(".isCheckingLogin", () => {
      describe("if true", () => {
        beforeEach(() => {
          props.isCheckingLogin = true;
        });
        it("should not render the app", () => {
          expect.assertions(1);
          const wrapper = shallow(<AppContainer {...props} />);
          expect(wrapper.find(App)).toHaveLength(0);
        });
      });
      describe("if false", () => {
        beforeEach(() => {
          props.isCheckingLogin = false;
        });
        it("should render the app", () => {
          expect.assertions(1);
          const wrapper = shallow(<AppContainer {...props} />);
          expect(wrapper.find(App)).toHaveLength(1);
        });
      });
    });
    describe("#onCheckLogin", () => {
      it("should be called when the component is rendered", () => {
        expect.assertions(1);
        shallow(<AppContainer {...props} />);
        expect(props.onCheckLogin.mock.calls).toHaveLength(1);
      });
    });
  });
});
