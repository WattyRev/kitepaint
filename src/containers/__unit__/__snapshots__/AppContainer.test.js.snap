// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders 1`] = `
<BrowserRouter>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "aqua": "#7FDBFF",
          "black": "#111111",
          "blue": "#0074D9",
          "blueDarker": "#005daa",
          "blueLighter": "#f0fbff",
          "fuchsia": "#F012BE",
          "gray": "#AAAAAA",
          "grayDark": "#5e5e5e",
          "grayDarker": "#3a3a3a",
          "green": "#2ECC40",
          "lime": "#01FF70",
          "maroon": "#85144B",
          "navy": "#001F3F",
          "olive": "#3D9970",
          "orange": "#FF851B",
          "purple": "#B10DC9",
          "red": "#FF4136",
          "silver": "#DDDDDD",
          "teal": "#39CCCC",
          "white": "#FFFFFF",
          "yellow": "#FFDC00",
        },
        "patterns": Object {
          "transparencyBackground": Array [
            "
  background-image: linear-gradient(45deg, ",
            "#AAAAAA",
            " 25%, transparent 25%),
    linear-gradient(-45deg, ",
            "#AAAAAA",
            " 25%, transparent 25%),
    linear-gradient(45deg, transparent 75%, ",
            "#AAAAAA",
            " 75%),
    linear-gradient(-45deg, transparent 75%, ",
            "#AAAAAA",
            " 75%);
  background-color: ",
            "#5e5e5e",
            ";
  background-size: 20px 20px;
  background-position: 0 0, 0 10px, 10px -10px, -10px 0px;
",
          ],
        },
      }
    }
  >
    <App
      _isEmbedded={false}
    />
  </ThemeProvider>
</BrowserRouter>
`;
